File: frontend/src/types/escalation.ts
export interface Escalation {
  id: number
  hospital_id: number
  patient_id: number
  queue_item_id: number | null
  call_id: number | null
  consensus_assessment_id: number | null

  reason: string
  priority: string
  evidence: string | null

  status: string
  source: string

  final_action: string | null

  resolved_at: string | null
  resolution_notes: string | null

  created_at: string
  updated_at: string
}

export type EscalationPriority = "CRITICAL" | "HIGH" | "MEDIUM" | "LOW"
export type EscalationStatus = "OPEN" | "IN_REVIEW" | "RESOLVED" | "CLOSED"

export interface EscalationStatusUpdate {
  status: EscalationStatus
  resolution_notes?: string
}

File: backend/schemas/escalation.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class EscalationStatusUpdate(BaseModel):
    status: str
    resolution_notes: str | None = None


class EscalationResponse(BaseModel):
    id: int

    hospital_id: int
    patient_id: int

    queue_item_id: int | None
    call_id: int | None
    consensus_assessment_id: int | None

    reason: str
    priority: str
    evidence: dict | None

    status: str
    source: str

    final_action: str | None

    created_at: datetime
    updated_at: datetime | None

    resolved_at: datetime | None
    resolution_notes: str | None

    model_config = ConfigDict(
        from_attributes=True
    )
